dissect/target/plugins/os/unix/log/lastlog.py

```diff
@@ -39,27 +66,27 @@ def lastlog(self):
             - https://www.tutorialspoint.com/unix_commands/lastlog.htm
         """
         try:
-            wtmp = self.target.fs.open("/var/log/lastlog")
+            lastlog = self.target.fs.open("/var/log/lastlog")
         except FileNotFoundError:
             return
 
-        log = UtmpFile(wtmp)
+        users = dict()
+        for user in self.target.users():
+            users[user.uid] = user.name
+
+        log = LastLogFile(lastlog)
 
-        for entry in log:
+        for idx, entry in enumerate(log):
 
-            if entry.ut_type in utmp.Type.reverse:
-                r_type = utmp.Type.reverse[entry.ut_type]
-            else:
-                r_type = None
+            # if ts=0 the uid has never logged in before
+            if entry.ut_host.decode().strip("\x00") == "" or entry.ll_time.tv_sec == 0:
+                continue
 
             yield LastlogRecord(
-                ts=datetime.datetime.utcfromtimestamp(entry.ut_tv.tv_sec),
-                ut_type=r_type,
-                ut_pid=entry.ut_pid,
-                ut_user=entry.ut_user.decode().strip("\x00"),
-                ut_line=entry.ut_line.decode().strip("\x00"),
-                ut_id=entry.ut_id.decode().strip("\x00"),
-                ut_host=entry.ut_host.decode().strip("\x00"),
-                ut_addr=socket.inet_ntoa(struct.pack("<i", entry.ut_addr_v6[0])),
+                ts=datetime.datetime.utcfromtimestamp(entry.ll_time.tv_sec),
```

Comment:
Preferably use:
```
from dissect.util import ts

ts.from_unix(entry.ll_time.tv_sec)
```
Since this method correctly sets the timezone information on the resulting `datetime` object.